It still errors for me:

    $ make
    [  2%] Built target pcre
    [  5%] Built target ljpeg
    [  6%] Built target zlib
    [  7%] Built target tinyxml
    [ 20%] Built target collada
    [ 20%] Built target glad
    [ 30%] Built target SDL2
    [ 32%] Built target opcode
    [ 34%] Built target recast
    [ 34%] Built target squish
    [ 35%] Built target convexDecomp
    [ 37%] Built target libtheora
    [ 37%] Built target libogg
    [ 38%] Built target libvorbis
    [ 38%] Built target nativeFileDialogs
    [ 39%] Built target lpng
    [ 39%] Building CXX object Tools/CMake/CMakeFiles/BaseGame.dir/__/__/Engine/source/platformX86UNIX/x86UNIXFileio.cpp.o
    /data/programming/learning/torque3d/Torque3D/Engine/source/platformX86UNIX/x86UNIXFileio.cpp: In function ‘const char* Platform::getExecutablePath()’:
    /data/programming/learning/torque3d/Torque3D/Engine/source/platformX86UNIX/x86UNIXFileio.cpp:962:20: error: ‘torque_getexecutablepath’ was not declared in this scope
            if( (cpath = torque_getexecutablepath()) )
                        ^~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminated due to -Wfatal-errors.
    make[2]: *** [Tools/CMake/CMakeFiles/BaseGame.dir/build.make:12595: Tools/CMake/CMakeFiles/BaseGame.dir/__/__/Engine/source/platformX86UNIX/x86UNIXFileio.cpp.o] Error 1
    make[1]: *** [CMakeFiles/Makefile2:138: Tools/CMake/CMakeFiles/BaseGame.dir/all] Error 2
    make: *** [Makefile:130: all] Error 2

I'm using Arch Linux 64 bit.

--- a/Engine/source/platformX86UNIX/x86UNIXFileio.cpp
+++ b/Engine/source/platformX86UNIX/x86UNIXFileio.cpp
@@ -55,7 +55,7 @@
  #include "console/console.h"
  #include "core/strings/stringFunctions.h"
  #include "util/tempAlloc.h"
- #include "cinterface/cinterface.h"
+ #import "cinterface/c_controlInterface.h"
  #include "core/volume.h"
 
  #if defined(__FreeBSD__)

--- a/Engine/source/platformX86UNIX/x86UNIXFileio.cpp
+++ b/Engine/source/platformX86UNIX/x86UNIXFileio.cpp
@@ -55,7 +55,7 @@
  #include "console/console.h"
  #include "core/strings/stringFunctions.h"
  #include "util/tempAlloc.h"
- #import "cinterface/c_controlInterface.h"
+ #include "cinterface/c_controlInterface.h"
  #include "core/volume.h"
 
  #if defined(__FreeBSD__)

